src/Build/BuildCheck/Infrastructure/CustomConfigurationData.cs

```diff
@@ -58,7 +68,11 @@ public override bool Equals(object? obj)
         return Equals((CustomConfigurationData)obj);
     }
 
-    protected bool Equals(CustomConfigurationData other) => Equals(ConfigurationData, other.ConfigurationData);
+    protected bool Equals(CustomConfigurationData other) {
+        // TODO: update the comparison: Compare ruleID, and exact match of the configuration data
```

Comment:
@JanKrivanek FYI this one is left, hence the validaty of the custom configuration per file is not checked at the moment.

Reply:
I'll add this in the next iterations of the review process. The test is prepared :)

Reply:
Update the Equals logic + added testing